compiler/scopelang/from_desugared.mli

```diff
@@ -16,4 +16,10 @@
 
 (** Translation from {!module: Desugared.Ast} to {!module: Scopelang.Ast} *)
 
-val translate_program : Desugared.Ast.program -> Shared_ast.untyped Ast.program
+val translate_program :
```

Comment:
Not at all sure about this, but if it can be decoupled in two steps, would it be feasible to have a first function that generates the exception graph, then the `translate_program` taking that graph as parameter ?

This would be more consistent with the way `Desugared.from_surface` works, taking as parameter a name resolution context that has been previously computed. Not sure it's better in general though ?